dashboard/utils.py

```diff
@@ -292,35 +283,139 @@ def attendance_list(request, event):
+    :returns:
+        the start date of the semester
+    :rtype: datetime
+
+    """
+    # Find the start month for the semester season
+    start_month = {
+        '0': 1,
+        '1': 6,
+        '2': 8
+    }.get(season)
+
+    start_day_of_month = 1
+
+    return datetime.datetime(year, start_month, start_day_of_month)
+
+
+def create_recurring_events(begin_date, day, interval, event_creator):
+    """Create recurring events for the semester begin_date falls in.
+
+    :param datetime begin_date:
+        The earliest date at which an event could be scheduled.  Events are
+        schedule only during the same semester as begin_date.
+    :param int day:
+        The integer day of the week on which the recurring events happen
+    :param int interval:
+        The number of day in between each event
+    :param function event_creator:
+        A function that takes (datetime, Semester) and creates an event.  The
+        function does not need to return a value (any value returned is ignored).
+
+    """
+    date = begin_date
+    semester = get_semester(date)
+
+    # offset to the next week
+    if date.weekday() >= day:
+        date_offset = 7 + day - date.weekday()
+    # otherwise use this week
+    elif date.weekday() < day:
+        date_offset = day - date.weekday()
+
     date = date + datetime.timedelta(days=date_offset)
     start_date = date
-    end_date = date
-    if semester.season == '2':
-        end_date = datetime.datetime(date.year, 12, 31)
-    elif semester.season == '0':
-        end_date = datetime.datetime(date.year, 5, 31)
-    day_count = int((end_date - start_date).days / instance['meeting_interval']) + 1
+    end_date = semester_end_date(semester.season, date.year)
+
+    day_count = int((end_date - start_date).days / interval) + 1
+    for date in (start_date + datetime.timedelta(interval) * n for n in range(day_count)):
+        event_creator(date, semester)
+
+
+def create_committee_event(date, semester, meeting_time, committee_object):
+    """Create a single committee meeting event for the given committee
+
+    :param datetime date:
+        The date of the committee meeting
+    :param Semester semester:
+        The semester in which the event occurs
+    :param datetime.time meeting_time:
+        The time of the day at which the meeting will occur
+    :param Committee committee_object:
+        The committee for which the event belongs to
+
+    """
+    event = CommitteeMeetingEvent(date=date, start_time=meeting_time, semester=semester,
+                                  committee=committee_object, recurring=True)
+    event.save()
+    event.eligible_attendees.set(committee_object.members.order_by('last_name'))
+    event.save()
+
+
+def create_recurring_meetings(instance, committee):
+    """Create recurring committee meetings for the given committee
+
+    :param dict instance:
+        a dictionary that defines the day of the week in 'meeting_day' on which meetings
```

Comment:
It might be important to note that it's not just any dictionary, it's a dictionary that's resulting from the return value of form.cleaned_data. As such, the keys are actually the fields of a form, namely the CommitteeCreateForm.